### PR TITLE
feat(solid-query): Solid Query Adapter For TanStack Query

### PR DESCRIPTION
**Pull request by ardeora**
This pull request aims to add the TanStack Query adapter for Solid JS. This pull request will add the following key primitives to the package.

- `createQuery`
- `createQueries`
- `createInfiniteQueries`
- `createMutation`
- `useIsFetching`
- `useIsMutating`
- `useQueryClient`
- `QueryClient`
- `QueryClientProvider`

The adapter docs have been updated as well as examples have been added. Please let me know how I can help get this massive PR accepted in the `beta` branch

### Acknowledgments
The work on solid-query could not have been possible without
- @lukesmurray
- @jennyckaplan

### CODE REVIEW

File: examples/solid/default-query-function/package-lock.json [hunk 1]
```diff
@@ -0,0 +1,1737 @@
+{
```
Comment (TkDodo) [inline examples/solid/default-query-function/package-lock.json:1]: I think there should only be one `package-lock.json` file at the root of the monorepo. The examples should be listed as a workspace here:

https://github.com/TanStack/query/blob/ea80700c2c8d48c6d0cb4ae0165a424ebea5d35c/package.json#L30-L56

**Review by TkDodo — COMMENTED** [review]
wow, this is amazing 🚀 

I don't know enough about solid to comment on specifics, but the react-query parts are really awesome ❤️

**Comment by lukesmurray** [comment]
thanks @TkDodo. we can remove the lock file.

couple questions. 

do we need to do anything to get our examples listed on the website with codesandbox integration?

do we need all tests passing to get this merged and published on npm? currently there are two failing tests. 

lastly wanted to confirm any of the changes we made outside the solid-query and examples folders, for instance, changes to rollup.config.ts are ok before we merge.

**Comment by TkDodo** [comment]
> lastly wanted to confirm any of the changes we made outside the solid-query and examples folders, for instance, changes to rollup.config.ts are ok before we merge.

I don't know much about the rollup config, I think @DamianOsipiuk does :)

File: examples/solid/default-query-function/package-lock.json (hunk 1 above)
Comment (lukesmurray) [reply examples/solid/default-query-function/package-lock.json:1]: resolved in #4211.

**Comment by lukesmurray** [comment]
This PR can probably be closed and we can use #4211 instead where I rebased the changes onto `main`.

**Pull request closed without merging**